Refresh Pro contact us form

britneywwc opened the pull request:
## Done

- Apply redesign on `__pro-contact-us-form.html`
- Ensure that the form payload submits to Marketo successfully
- Added checkbox validation and toggle checkbox visibility functions to dynamic forms 
- Fix missing form payload error

## QA

- Check out these forms:
  - [/contact-us/form?product=pro](https://ubuntu-com-14124.demos.haus/contact-us/form?product=pro)
  - [/pricing/pro#get-in-touch](https://ubuntu-com-14124.demos.haus/pricing/pro#get-in-touch)
- Check that the fields are similar to the ones on prod: https://ubuntu.com/contact-us/form?product=pro
- Fill up form and submit
- Check that payload reflects input correctly

## Issue / Card

Fixes [WD-13105](https://warthogs.atlassian.net/browse/WD-13105)

## Screenshots

[If relevant, please include a screenshot.]


## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)


[WD-13105]: https://warthogs.atlassian.net/browse/WD-13105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14124.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/7809)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

akbarkz commented:
@britneywwc two findings from QA:
1. When opening https://ubuntu-com-14124.demos.haus/pricing/pro#get-in-touch, clicking on X on the top right corner does not close the modal.
2. On that same form the field for phone does not use `intl-tel-input` package as on other forms.

britneywwc commented:
@akbarkz this is ready for your review. As I was addressing the issues you raised on the dynamic forms, I went ahead and added some logic on `dynamic-forms.js` which should address it and also fix the previously missing payload on dynamic forms.
P.S. The failing lint-python CI action can be ignored as this will be addressed on the feature branch once it's rebased to main.

@lyubomir-popov can you please have a quick look at the design of the [dynamic form](https://ubuntu-com-14124.demos.haus/pricing/pro#get-in-touch)? I implemented it similar to the design on our [static forms](https://ubuntu-com-14124.demos.haus/contact-us/form?product=pro).

Thanks!

akbarkz commented:
@britneywwc Prettier is not liking something in `dynamic-forms.js`, please check
